### PR TITLE
Update and refactor Cypress tests

### LINKED ISSUE #1450: [Cypress] Add 2021 Data Publication tests
## Tests
  - [ ] Data Publications shows correct reports for each year
  - [ ] One Year 
    - [ ] Shows 2017, 2018, 2019(?)
    - [ ] Files exist
  - [ ] Three Year
    - [ ] Shows 2017, 2018
    - [ ] Files exist

### LINKED ISSUE #1460: Fix Cypress tests
- [x] NationalAggregateReports.spec
- [x] AggregateReports.spec
- [x] DisclosureReports.spec
- [x] maps.spec

### PR DESCRIPTION
**Pull request by meissadia**
Closes #1460 
Closes #1450 

## Changes

- Update Publication Report tests to account for auto-redirect when visiting their base url
- Add 2020 and 2021 tests for Data Browser: Dataset Filtering & Maps
- Update 2018 and 2019 Maps tests to account for updated values due to recent data publication release
- Consolidate per-year Dataset Filtering tests into a single file `cypress/data-browser/dataset-filtering.spec.js`, since they were 99% duplicated code.

## Testing

1. Proxy to Production
2. `yarn cypress open`
3. Run tests for Reports, Data Browser

### CODE REVIEW

File: cypress/integration/data-browser/dataset-filtering.spec.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  isBeta,
+  isProd,
+  waitUpto2Mins,
+} from '../../support/helpers'
+
+const YEARS = [2021, 2020, 2019, 2018]
+const { HOST, ENVIRONMENT } = Cypress.env()
+const dbUrl = dbURL.bind(null, HOST)
+
+describe(`Data Browser - Dataset Filtering`, function () {
```
Comment (lchen-2101) [inline cypress/integration/data-browser/dataset-filtering.spec.js:13]: newer standards is to have arrow functions now right? think keyword function is only useful for prototypal functionalities now.

File: cypress/integration/data-publication/AggregateReports.spec.js [hunk 2]
```diff
@@ -1,88 +1,166 @@
-import { openSelector } from '../../support/helpers'
+import { openSelector } from "../../support/helpers"
```
Comment (lchen-2101) [inline cypress/integration/data-publication/AggregateReports.spec.js:1]: quote changes are from eslint?

Comment (meissadia) [reply cypress/integration/data-publication/AggregateReports.spec.js:1]: Looks like it, that change is from @Michaeldremy.  We're working on unifying code conventions in https://github.com/cfpb/hmda-frontend/issues/1440.

**Review by lchen-2101 — APPROVED** [review]
LGTM
for future refactors / clean ups, maybe get rid of the template literals as well when there are no string interpolations.

**Pull request merged**